docs/src/pages/[...docPath].tsx

```diff
@@ -0,0 +1,130 @@
+import type { NextPage, GetStaticProps, GetStaticPaths } from "next";
+import fs from "fs";
+import matter from "gray-matter";
+import { MDXRemote } from "next-mdx-remote";
+import { serialize } from "next-mdx-remote/serialize";
+import remarkDirective from "remark-directive";
+import { visit } from "unist-util-visit";
+import { h } from "hastscript";
+
+import {
+  getMDPaths,
+  withIndexFile,
+  withIndexURL,
+  withInsertedCodeFromLinks,
+  withoutComments,
+} from "../model/md-generate";
+import Title from "../components/mdxComponents/Title";
+import Paragraph from "../components/mdxComponents/Paragraph";
+import CodeBlocks from "../components/mdxComponents/CodeBlocks";
+import Admonition from "../components/mdxComponents/Admonition";
+import DocumentationLayout from "../components/DocumentationLayout";
+
+const components = {
+  h2: Title.H2,
+  h3: Title.H3,
+  p: Paragraph,
+  code: CodeBlocks.Code,
+  pre: CodeBlocks.Pre,
+  tip: Admonition.Tip,
+  warning: Admonition.Warning,
+};
+
+/** @type {import('unified').Plugin<[], import('mdast').Root>} */
+function myRemarkPlugin() {
+  // @ts-ignore
+  return (tree) => {
+    visit(tree, (node) => {
+      if (
+        node.type === "textDirective" ||
+        node.type === "leafDirective" ||
+        node.type === "containerDirective"
+      ) {
+        const data = node.data || {};
+        const hast = h(node.name, node.attributes);
+        // @ts-ignore
+        data.hName = hast.tagName;
+        // @ts-ignore
+        data.hProperties = hast.properties;
+      }
+    });
+  };
+}
+
+interface IFrontMatter {
+  title: string;
+  description: string;
+  anchors?: string[];
+}
+interface IDocPage {
+  source: string;
+  frontMatter: IFrontMatter;
+}
+
+const DocPage: NextPage<IDocPage> = ({ source, frontMatter }): JSX.Element => {
+  return (
+    <DocumentationLayout
+      seo={{
+        title: frontMatter.title
+          ? frontMatter.title.concat(" | Hardhat")
+          : "Hardhat",
+        description:
+          frontMatter.description ||
+          "Ethereum development environment for professionals by Nomic Foundation",
+      }}
+    >
+      {/* @ts-ignore */}
+      <MDXRemote {...source} components={components} />;
+    </DocumentationLayout>
+  );
+};
+
+export default DocPage;
+
+export const getStaticProps: GetStaticProps = async (props) => {
```

Comment:
I didn't check - is it possible to specify props types here?

Reply:
I've tried and've got next error:

```
 Types of property 'params' are incompatible.
        Type 'ParsedUrlQuery | undefined' is not assignable to type '{ [key: string]: any; }'.
          Type 'undefined' is not assignable to type '{ [key: string]: any; }
         
```